packages/common/src/channel-list/channel-list.tsx

```diff
@@ -28,7 +27,7 @@ export interface ChannelListProps {
  * modal with channels available to join. Make sure to handle the onChannelSwitched event to switch
  * the current channel passed to the Chat provider.
  */
-export const ChannelList: FC<ChannelListProps> = (props: ChannelListProps) => {
+export const useChannelListCore = (props: CommonChannelListProps) => {
```

Comment:
I'm not sure if I like changing from a component to a hook while the name of the file remains as it was. Why is it not in /hooks anyway?